response.go

```diff
@@ -0,0 +1,24 @@
+package sling
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+// ResponseDecoder is a generic interface for abstracting away the implementation details
+// of decoding the API response.
+// See reference implementation of jsonDecoder for an example.
+// Attention: the *caller* is expected to close the response body.
```

Comment:
You can drop this, it should be an implementation detail since some some implementation might close the body and prevent further reads.